pages/en/lb4/Defining-the-API-using-bottom-up-approach.md

```diff
@@ -0,0 +1,156 @@
+---
+lang: en
+title: 'Defining the API using bottom up approach'
+keywords: LoopBack 4.0, LoopBack 4
+tags:
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Defining-the-API-using-bottom-up-approach.html
+summary:
+---
+
+## Define the API from bottom to top
+
+### Start with LoopBack artfiacts
+
+Sometimes, it's hard to know what your API is going to look like without
+writing out your application first. It also might be difficult to adjust your
+API as plans change or as features need to be reworked.
```

Comment:
IMO, the reasoning for bottom-up is misleading here. Please rewrite it to align with some ideas below:

1. Designing APIs in vacuum is not only difficult but also not wise if you don't have a good understanding of what your existing system can offer and what your clients need.

2. Bottom-up approach allows developers to capture existing business domain models and corresponding operations and reflect them for external consumption as APIs. Such information can be defined as metadata with code, JSON/YAML documents, or decorations.

3. Bottom-up approach better aligns APIs with implementations from the beginning and evolve them all together as API consumers and providers reach agreements incrementally.

4. Bottom-up approach makes it possible to have APIs up and running for play and polish at very early stages, which is critical to the success of APIs as stakeholders can see the big picture collaboratively before it's too late.